### DISCUSSION TITLE
GATT characteristic long writes

### DISCUSSION THREAD

**Discussion by AndoniV** [ThomasGerstenberg/blatann#136] [Q&A] [answered]
Any idea how to perform long write/reads to a characteristic? I can see an object `blatann.nrf.nrf_types.enums.BLEGattWriteOperation(value)` that defines the type of write request. I'm interested in the `prepare_write_req` and `execute_write_req` values.

Any help would be very welcome!

**Comment by ThomasGerstenberg** [comment, marked as answer]
Long read/writes are automatically handled within blatann. When reading from a characteristic, if the length of the data is equal to the MTU size it'll automatically perform a long read in order to check if there's more data on the characteristic. This is equivalent to how Android/iOS handle characteristic reads.

Same goes for long writes. If you write data larger than an MTU to a characteristic (given the characteristic is large enough), it'll automatically perform long writes

**Reply by AndoniV** [reply]
Understood, thank you very much for your reply really appreciate it. Will try this out now!
